packages/flutter/lib/src/widgets/animated_fade_out_fade_in.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'implicit_animations.dart';
+import 'transitions.dart';
+
+class AnimatedFadeOutFadeIn extends ImplicitlyAnimatedWidget {
```

Comment:
I don't think we should make this public as it isn't really re-usable outside of the `FadeInImage` given its targeted API.

Reply:
Sounds good 👍 I'll bring it back in.

Reply:
Thinking about it a bit more, I think we should extract this if we want to support a way to delegate out the animation. If we don't then I don't think we can do:

```dart
FadeInImage(
    placeholder: MemoryImage(bytes),
    image: NetworkImage('https://backend.example.com/image.png'),
    animationBuilder: (Widget target, Widget placeholder, bool isTargetLoaded) => AnimatedFadeOutFadeIn(
        target: target,
        placeholder: placeholder,
        isTargetLoaded: isTargetLoaded,
        fadeInDuration: const Duration(milliseconds: 300),
        fadeOutDuration: Curves.easeOut,
        fadeInCurve: const Duration(milliseconds: 700),
        fadeOutCurve: Curves.easeIn,
    );
)
```

as `_AnimatedFadeOutFadeIn` isn't public in current-master?